xapian-core/include/xapian/queryparser.h

```diff
@@ -389,6 +389,48 @@ class XAPIAN_VISIBILITY_DEFAULT NumberRangeProcessor : public RangeProcessor {
     Xapian::Query operator()(const std::string& begin, const std::string& end);
 };
 
+/** Handle a byte unit range.
+ *
+ *  This class must be used on values which have been encoded using
+ *  Xapian::sortable_serialise() which turns numbers into strings which
+ *  will sort in the same order as the numbers (the same values can be
+ *  used to implement a numeric sort).
+ */
+class XAPIAN_VISIBILITY_DEFAULT UnitRangeProcessor : public RangeProcessor {
+  public:
+    /** Constructor.
+     *
+     *  @param slot_    The value slot number to query.
+     *
+     *  @param str_     A string to look for to recognise values as belonging
+     *                  to this numeric range.
+     *
+     *  The string supplied in str_ is the prefix plus ":" that defines the field
+     *  upon which the range query is performed.
+     *
+     *  For example, if str_ is "size:", and the range
+     *  processor has been added to the queryparser, the queryparser will
+     *  accept "size:3K..10K" as a valid range.
+     */
+    UnitRangeProcessor(Xapian::valueno slot_,
+			 const std::string &str_ = std::string())
```

Comment:
This line's indented two columns too far (though the diff viewer makes it look like it's just one column too far).

Reply:
In NumberRangeProcessor's constructor, the c of const is aligned with X of Xapian::valueno, if I'm not wrong. If that's the case then I believe second line above is only one column too far. I've added a new commit moving back one column.

Reply:
The alignment here is now good, except it's using all spaces rather than being tab-filled, as `xapian-check-patch` points out:

```
xapian-core/include/xapian/queryparser.h:416: error: added/changed line uses spaces for indentation rather than tab: +                       const std::string &str_ = std::string())
```

The Xapian convention for C and C++ code is that the indent is done with a series of 0 or more tabs (each 8 columns wide), then a series of 0 or more spaces (each one column wide), using as many tabs as possible.

The difference isn't visible in many cases, but there are benefits to having a canonical representation when there's no visible difference - especially it means that you can't have two files which look the same but aren't, or at least if you do one of them must be "wrong".  That's very helpful when dealing with merging changes, because if everything follows the rules then you avoid merge conflicts due to whitespace changes in one version.

For similar reasons, source file lines shouldn't have trailing whitespace, and the final line of the file should end in a newline, but there shouldn't be further empty lines after it.

Reply:
Thanks for clarification regarding the indentation style; pushed fix-up commit.